src/festim/coupled_heat_hydrogen_problem.py

```diff
@@ -0,0 +1,200 @@
+    Examples:
+        .. highlight:: python
+        .. code-block:: python
+
+            import festim as F
+
+            my_heat_transfer_model = F.HeatTransferProblem(
+                mesh=F.Mesh(...),
+                subdomains=[F.Subdomain(...)],
+                ...
+            )
+
+            my_h_transport_model = F.HydrogenTransportProblem(
+                mesh=F.Mesh(...),
+                subdomains=[F.Subdomain(...)],
+                species=[F.Species(name="H"), F.Species(name="Trap")],
+                ...
+            )
+
+            coupled_problem = F.CoupledHeatTransferHydrogenTransport(
+                heat_problem=my_heat_transfer_model,
+                hydrogen_problem=my_h_transport_model,
+            )
+
+
+    """
+
+    heat_problem: HeatTransferProblem
+    hydrogen_problem: HydrogenTransportProblem
+
+    non_matching_meshes: bool
+
+    def __init__(
+        self,
+        heat_problem: HeatTransferProblem,
+        hydrogen_problem: HydrogenTransportProblem,
+    ) -> None:
+        self.heat_problem = heat_problem
+        self.hydrogen_problem = hydrogen_problem
+
+    @property
+    def heat_problem(self):
+        return self._heat_problem
+
+    @heat_problem.setter
+    def heat_problem(self, value):
+        if not isinstance(value, HeatTransferProblem):
+            raise TypeError("heat_problem must be a festim.HeatTransferProblem object")
+        self._heat_problem = value
+
+    @property
+    def hydrogen_problem(self):
+        return self._hydrogen_problem
+
+    @hydrogen_problem.setter
+    def hydrogen_problem(self, value):
+        if not isinstance(value, HydrogenTransportProblem):
+            raise TypeError(
+                "hydrogen_problem must be a festim.HydrogenTransportProblem object"
+            )
+        self._hydrogen_problem = value
+
+    @property
+    def non_matching_meshes(self):
+        return self.heat_problem.mesh.mesh != self.hydrogen_problem.mesh.mesh
+
+    def initialise(self):
+        if (
+            self.heat_problem.settings.transient
+            and self.hydrogen_problem.settings.transient
+        ):
+            # make sure both problems have the same initial time step and final time,
+            # use minimal initial value of the two and maximal final time of the two
+            min_initial_dt = min(
+                float(self.heat_problem.settings.stepsize.initial_value),
+                float(self.hydrogen_problem.settings.stepsize.initial_value),
+            )
+            self.heat_problem.settings.stepsize.initial_value = min_initial_dt
+            self.hydrogen_problem.settings.stepsize.initial_value = min_initial_dt
+
+            max_final_time = max(
+                float(self.heat_problem.settings.final_time),
+                float(self.hydrogen_problem.settings.final_time),
+            )
+            self.heat_problem.settings.final_time = max_final_time
+            self.hydrogen_problem.settings.final_time = max_final_time
```

Comment:
I think we shouldn't allow different final times. What do you think?

Reply:
Sure